src/DetailView/DetailActions.js

```diff
@@ -127,3 +140,104 @@ export function cancelPollData() {
     dispatch({type: CANCEL_POLLING_DATA});
   };
 }
+
+function updateSuccess() {
+  return dispatch => {
+    dispatch(fetchData());
+  };
+}
+
+function updateError(error) {
+  if (error.response) {
+    switch (error.response.status) {
+      case 400:
+        error = 'Invalid request.';
+        break;
+      case 404:
+        error = 'Resource has not been found.';
+        break;
+      case 409:
+        error = 'Conflict.';
+    }
+  } else {
+    error = error.toString();
+  }
+  return dispatch => {
+    dispatch({type: UPDATE_FAILURE, error});
+  };
+}
+
+export function updateData(data) {
+  return (dispatch, getState) => {
+    const state = getState();
+    const {url} = state.detailReducer;
+    const {url: gohanUrl} = state.configReducer.gohan;
+    const {tokenId} = state.authReducer;
+    const headers = {
+      'Content-Type': 'application/json',
+      'X-Auth-Token': tokenId
+    };
+
+    axios.put(gohanUrl + url, data, {headers}).then(response => {
+      const {status} = response;
+
+      if (status === 200) {
+        dispatch(updateSuccess());
+      } else {
+        dispatch(updateError('Cannot update resource!'));
+      }
+    }).catch(error => {
+      dispatch(updateError(error));
+    });
+  };
+}
+
+function deleteSuccess() {
+  browserHistory.goBack();
+  return dispatch => {
+    dispatch({type: DELETE_SUCCESS});
+  };
+}
+function deleteError(error) {
+  if (error.response) {
+    switch (error.response.status) {
+      case 400:
+        error = 'Invalid request.';
+        break;
+      case 404:
+        error = 'Resource has not been found.';
+        break;
+      case 409:
+        error = 'Conflict.';
+    }
+  } else {
+    error = error.toString();
+  }
```

Comment:
It looks the same now, but this is where custom error messages are handled for each action type.